lib/matplotlib/image.py

```diff
@@ -603,11 +603,12 @@ def contains(self, mouseevent):
             xmin, xmax = xmax, xmin
         if ymin > ymax:
             ymin, ymax = ymax, ymin
-
-        if x is not None and y is not None:
-            inside = (xmin <= x <= xmax) and (ymin <= y <= ymax)
-        else:
-            inside = False
+  
+        trans = self.get_transform()
+        bbox = Bbox(np.array([[xmin, ymin], [xmax, ymax]]))
+        transformed_bbox = TransformedBbox(bbox, trans)
+       
+        inside = transformed_bbox.contains(mouseevent.x,mouseevent.y)
```

Comment:
@jklymak Sounds very reasonable.  Understanding the class well enough to find places at which the extents and transforms are modified is more than I have the capacity to take on at present, so regretfully, much as I would like to help further I am going to have to leave this hanging out there and hope that others might have the time and interest to carry to completion.

Reply:
@ilopata1 This solution may be the best we can do for now.  But it was a bit of an a-ha moment for why so many transforms get created even if a plot is not changing.

Does `contains` get called all the time, or just after a mouse click?  If just after a mouse click, its probably not at all a performance problem for this change to be put in.

Reply:
I can't speak to when/whether it may get called internally, but from an application perspective it is most likely to get called on mouse events. This may include mouse **moves**, which can be generated in high volume.

I just ran a test and on an Intel i5 processor, over 18 executions of this code fragment the average execution time was 0.00022 seconds.  To put that another way, contains would need to be called 4,500 times on my machine for the execution time to be one second.

Reply:
I think(?) you could apply the *inverse* transform to the event coordinates and check whether that's in the untransformed bbox?  That'll avoid creating a TransformedBbox, which per above is not so cheap.